src/fhir/binaries.ts

```diff
@@ -0,0 +1,36 @@
+import { Provenance } from "fhir/r4";
+import { CTWRequestContext } from "@/components/core/providers/ctw-context";
+import { find, some } from "@/utils/nodash";
+
+export function getBinaryId(
+  provenances: Provenance[],
+  targetId: string
+): string | undefined {
+  for (let i = 0; i < provenances.length; i += 1) {
+    const provenance = provenances[i];
+
+    const hasTarget = some(provenance.target, (t) =>
+      t.reference?.includes(targetId)
+    );
+
+    if (hasTarget) {
+      const source = find(provenance.entity, { role: "source" });
+      if (source) {
+        // Return the ID portion of the reference.
+        return source.what.reference?.split("/")[1];
```

Comment:
Is it possible that `source` is defined but `source.what.reference` is undefined? TypeScript seems to think so, but if it's a case of we know better than the TS engine feel free to resolve this convo.
```suggestion
      if (source?.what.reference) {
        // Return the ID portion of the reference.
        return source.what.reference?.split("/")[1];
```

Reply:
Yes that's better! TY! This way we'll also keep looping to try and find a binary with a valid `what.reference`.